dist/tools/compile_commands/compile_commands.py

```diff
@@ -33,21 +33,35 @@ def detect_includes_and_version_gcc(compiler):
             inputdata = b"typedef int dont_be_pedantic;"
             _, stderrdata = proc.communicate(input=inputdata)
     except FileNotFoundError:
-        msg = "Compiler {} not found, not adding system include paths\n".format(compiler)
+        msg = f"Compiler {compiler} not found, not adding system include paths\n"
         sys.stderr.write(msg)
         return []
 
     stderrdata = stderrdata.decode("utf-8")
     version = REGEX_VERSION.search(stderrdata).group(1)
-    includes = [os.path.abspath(p) for p in REGEX_INCLUDES.search(stderrdata).group(1).split()]
+    tmp = [os.path.abspath(p) for p in REGEX_INCLUDES.search(stderrdata).group(1).split()]
+
+    # the include path containing newlib.h must come prior to the gcc headers
+    # in the include list to avoid mischief from happening
+    newlib_path = ""
```

Comment:
Does this also work with picolibc? But I guess they just kept the name of the include.

Reply:
This special handling is not needed for picolibc. RIOT is not even passing any picolibc related includes to the compiler, all of this is handled fine by this `picolibc.specs`.

And this works just fine with `clangd` as well, as even with GCC the magic internal default include paths are overwritten when using `picolibc` anyway, as otherwise GCC would use the newlib system headers over the picolibc ones instead.